biz.aQute.bndlib/src/aQute/bnd/build/Project.java

```diff
@@ -2925,55 +3007,56 @@ public Jar pack(String profile) throws Exception {
 				return null;
 			}
 
-			Builder b = subBuilders.iterator()
-				.next();
-
-			ignore.remove(BUNDLE_SYMBOLICNAME);
-			ignore.remove(BUNDLE_VERSION);
-			ignore.add(SERVICE_COMPONENT);
-
-			try (ProjectLauncher launcher = getProjectLauncher()) {
-				launcher.getRunProperties()
-					.put("profile", profile); // TODO
-												// remove
-				launcher.getRunProperties()
-					.put(PROFILE, profile);
-				Jar jar = launcher.executable();
-				Manifest m = jar.getManifest();
-				Attributes main = m.getMainAttributes();
-				for (String key : this) {
-					if (Character.isUpperCase(key.charAt(0)) && !ignore.contains(key)) {
-						String value = getProperty(key);
-						if (value == null)
-							continue;
-						Name name = new Name(key);
-						String trimmed = value.trim();
-						if (trimmed.isEmpty())
-							main.remove(name);
-						else if (EMPTY_HEADER.equals(trimmed))
-							main.put(name, "");
-						else
-							main.put(name, value);
+			try (Builder b = subBuilders.iterator()
```

Comment:
It seems that calling pack on a project can be a bit non-deterministic since it always uses the first subbuilder but the first subbuilder is not well defined since the order of subbuilderscan be based upon File.listFiles() which is defined to not have a prescribed order of its results.

But oddly, this method only seems to use the subbuilder to obtain a BSN if there is no BSN in the launcher's executable jar...

Reply:
they were actually not needed, cleaned up